### ISSUE TITLE
Babel traverse failures while running tests on CircleCI

### ISSUE THREAD

**Issue by joeythomaschaske** [salesforce/sfdx-lwc-jest#189]
We're picking back up a Salesforce project after some time off.

We noticed our builds are failing due to test failures after November 20, 2020 with the error

```
Test suite failed to run

    TypeError: LWC1007: Cannot read property 'file' of undefined

      at Class (node_modules/@babel/plugin-proposal-class-properties/lib/index.js:454:24)
      at NodePath._call (node_modules/@babel/traverse/lib/path/context.js:55:20)
      at NodePath.call (node_modules/@babel/traverse/lib/path/context.js:42:17)
      at NodePath.visit (node_modules/@babel/traverse/lib/path/context.js:92:31)
      at TraversalContext.visitQueue (node_modules/@babel/traverse/lib/context.js:115:16)
      at TraversalContext.visitMultiple (node_modules/@babel/traverse/lib/context.js:79:17)
      at TraversalContext.visit (node_modules/@babel/traverse/lib/context.js:141:19)
      at Function.traverse.node (node_modules/@babel/traverse/lib/index.js:82:17)
      at traverse (node_modules/@babel/traverse/lib/index.js:64:12)
      at NodePath.traverse (node_modules/@babel/traverse/lib/path/index.js:142:24)
  ```
  
  This affects about 50% of our tests on circle ci running with orb `circleci/node:latest`.
  
  Running `npm list` shows a version  0.10.2 being installed
  
  ```
  /home/circleci/project
├── @salesforce/eslint-config-lwc@0.3.0
├── @salesforce/sfdx-lwc-jest@0.10.2
├── @salesforce/wire-service-jest-util@2.2.6
```

It looks like a [babel release](https://github.com/babel/babel/releases/tag/v7.12.7) on November 20, 2020 might be the culprit.

This is all the information I've been able to acquire so far.

Any idea on what might be causing these babel errors?

**Comment by pmdartus** [comment]
Thanks for reporting this issue. Since it's the first time such an issue is reported, could you share a minimal SFDX project with this issue? 

I am also wondering why timing is important here. Does your project use a lock file?

**Comment by joeythomaschaske** [comment]
Alright I was able to limit the scope of the problem to a custom mixin we use for handling form validation.

This is a dumbed down project with 2 LWC components. `accountName` is a basic component hosted on an account record page for updating the account name. It uses `ValidationMixin` to handle validation.

https://github.com/joeythomaschaske/lwc-jest-error

Running `npm run test` on the component included should reproduce the issue. If you comment out `[ValidationMixin.Selectors] = ['lightning-input'];` in `accountName.js` and rerun the tests they will pass.

Every test that is failing in our build uses this `ValidationMixin`

**Comment by pmdartus** [comment]
Thanks for the details repro @joeythomaschaske. I was able to reproduce the issue locally as well.

It appears, based on the initial debugging that I have done, that the babel plugin in charge of transforming class fields is in conflict with the LWC babel plugin. This is a part of the framework that we recently changed. It would require further investigation to better under the root cause.

The code path throwing an error is only invoked when a  class has computed class fields. Until this gets fixed, I would recommend moving `[ValidationMixin.Selectors]` in the constructor:

```js
export default class AccountName extends ValidationMixin(LightningElement) {
  constructor() {
    super();
    this[ValidationMixin.Selectors] = ['lightning-input'];
  }
}
```

**Comment by joeythomaschaske** [comment]
Thanks for the quick turn around!

**Comment by joeythomaschaske** [comment]
I pushed a new branch `workaround` with another test after moving the `[ValidationMixin.Selectors]`  into the constructor and it does not seem to be picking up the selectors that are set.

https://github.com/joeythomaschaske/lwc-jest-error/tree/workaround

It works in the browser, just not in jest.

**Comment by pmdartus** [comment]
This issue should have been solved with the latest release: `0.10.4`. If it doesn't work please delete your package-lock.json to make sure all the transitive dependencies get updated.

I will close this issue, for now, @joeythomaschaske. Feel free to comment if you are still experiencing some issues.
